### PR TITLE
Istioctl fixes to satisfy gometalinter

### PR DESCRIPTION
**Pull request by ymesika**

### CODE REVIEW

**Comment by ymesika** [comment]
I ran:
```sh
gometalinter.v2 --config=./lintconfig_base.json ./istioctl/...
```
Which is what `make lint` runs

File: istioctl/cmd/istioctl/metrics.go [hunk 1]
```diff
@@ -184,7 +184,10 @@ func availablePort() (int, error) {
 		return 0, err
 	}
 	port := l.Addr().(*net.TCPAddr).Port
-	l.Close()
+	err = l.Close()
+	if err != nil {
```
Comment (geeknoid) [inline istioctl/cmd/istioctl/metrics.go]: Combin with previous line.

**Comment by gyliu513** [comment]
@ymesika ah, I see, but my `make lint` failed, this seems to be a bug, do we still beed vet here?

```
cicd:~/go/src/istio.io/istio# go get golang.org/x/tools/cmd/vet
package golang.org/x/tools/cmd/vet: cannot find package "golang.org/x/tools/cmd/vet" in any of:
	/usr/local/go1.10.1/src/golang.org/x/tools/cmd/vet (from $GOROOT)
	/root/go/src/golang.org/x/tools/cmd/vet (from $GOPATH)
```

**Pull request merged**